### PR TITLE
MF-132: Display selected location in Navigation Header

### PR DESCRIPTION
**Pull request by ghost**
Displayed current session location in header
Used location svg from style guide
Set up jest-dom for testing elements in Document
Added test

### CODE REVIEW

File: src/root.component.test.tsx [hunk 1]
```diff
@@ -53,11 +55,29 @@ describe(`<Root />`, () => {
...
+      return Promise.resolve({});
+    });
     let wrapper = renderWithRouter(<Root />);
     fireEvent.click(wrapper.getByText("testuser"));
     await wait(() => {
       const logoutBtn = wrapper.getByText(/Logout/i);
       fireEvent.click(logoutBtn);
+      expect(mockOpenMrsFetch).toHaveBeenCalledWith("/ws/rest/v1/session", {
+        method: "DELETE"
+      });
```
Comment (brandones) [inline src/root.component.test.tsx:68]: Actually testing something! Huge improvement to this test, thank you :grinning:

File: src/root.component.tsx [hunk 2]
```diff
@@ -58,6 +59,15 @@ export function Root(props: NavProps) {
     }
   });
 
+  React.useEffect(() => {
+    const abortController = new AbortController();
+    openmrsFetch("/ws/rest/v1/appui/session", {
```
Comment (brandones) [inline src/root.component.tsx:64]: Isn't `sessionLocation` included in data returned from `getCurrentUser`?

**Comment by brandones** [comment]
Can we get a screenshot?

File: src/root.component.tsx (hunk 2 above)
Comment (ghost) [reply src/root.component.tsx:64]: No. getCurrentUser only returns data specific to logged in user only. Location is included as part of session.

**Comment by ghost** [comment]
Header screenshots:
image

image

File: src/root.component.tsx (hunk 2 above)
Comment (brandones) [reply src/root.component.tsx:64]: Right, I wish the naming for `getCurrentUser` was better, but it does return the whole session object when parameter `includeAuthStatus: true` is provided. Try adding `console.log` to the `getCurrentUser({ includeAuthStatus: true }).subscribe(` callback. You should see `response.sessionLocation`.

Comment (brandones) [reply src/root.component.tsx:64]: The typescript type of that function's return value might be incomplete, also.

Comment (ghost) [reply src/root.component.tsx:64]: Sure. I will check that.

Comment (ghost) [reply src/root.component.tsx:64]: It return something like this. It doesn't have location:
allowedLocales: (6) ["en", "en_GB", "es", "fr", "it", "pt"]
authenticated: true
locale: "en_GB"
sessionId: "7D7BD4DF9E550886B497D6CF31A408"
user:
display: "admin"
links: (2) [{…}, {…}]
person: {uuid: "24252571-dd5a-11e6-9d9c-0242ac1500", display: "Super User", links: Array(1)}
privileges: []
resourceVersion: "1.8"
retired: false
roles: (2) [{…}, {…}]
systemId: "admin"
userProperties: {loginAttempts: "0"}
username: ""
uuid: "45ce6c2e-dd5a-11e6-9d9c-0242ac1500"

Comment (brandones) [reply src/root.component.tsx:64]: Huh! That is different from what I remember but it looks like you're right. Thanks for investigating.

File: src/root.component.tsx [hunk 3]
```diff
@@ -58,6 +59,15 @@ export function Root(props: NavProps) {
...
+      signal: abortController.signal
+    }).then(({ data }) => {
+      setCurrentSession(data);
+    });
+  });
```
Comment (brandones) [inline src/root.component.tsx]: I think this will run on every render, no?
```suggestion
  }, []);
```

**Review by brandones — APPROVED** [review]
Thanks for doing this!

**Pull request merged**